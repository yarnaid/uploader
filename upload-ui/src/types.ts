import * as z from "zod";

export const filterSchema = z.object({
  name: z.string(),
  values: z.array(z.string()),
});
export const filtersSchema = z.array(filterSchema);

export type FilterType = z.infer<typeof filterSchema>;
export type FiltersType = z.infer<typeof filtersSchema>;

export const fileMetadataSchema = z.object({
  name: z.string().nullable().optional(),
  type_: z.string().nullable().optional(),
  size: z.number().nullable().optional(),
  created_at: z.string().nullable().optional(),
  updated_at: z.string().nullable().optional(),
  tags: z.array(filterSchema).nullable().optional(),
  file: z.string().nullable().optional(),
});

export type FileMetadataType = z.infer<typeof fileMetadataSchema>;

export type UploadFileRequest = {
  file?: File;
  raw_metadata?: string;
};
