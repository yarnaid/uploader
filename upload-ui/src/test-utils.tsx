import React, { ReactElement } from "react";
import { Provider } from "react-redux";
import { render, RenderOptions } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import configureMockStore from "redux-mock-store";
import { configureStore } from "@reduxjs/toolkit";
import rootReducer from "./redux/reducers";
import {
  FileMetadataType,
  FilterType,
  FileType,
  UploadFileType,
} from "./types";
import type { PreloadedState } from "@reduxjs/toolkit";
import type { RootState, AppStore } from "./redux/store";

// This type interface extends the default options for render from RTL, as well
// as allows the user to specify other things such as initialState, store.
interface ExtendedRenderOptions extends Omit<RenderOptions, "queries"> {
  preloadedState?: PreloadedState<RootState>;
  store?: AppStore;
}

const mockStore = configureMockStore();

const initialState: {
  files: { files: FileType[] };
  filters: FilterType[];
  file: string;
  tags: FilterType[];
  metadata: FileMetadataType;
} = {
  files: { files: [] },
  filters: [],
  file: "",
  tags: [],
  metadata: { tags: [] },
};

export const store: any = mockStore(initialState);

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <Provider store={store}>
      <MemoryRouter>{children}</MemoryRouter>
    </Provider>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = configureStore({ reducer: rootReducer, preloadedState }),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) => render(ui, { wrapper: AllTheProviders, ...options });
export * from "@testing-library/react";

export { customRender as render };
