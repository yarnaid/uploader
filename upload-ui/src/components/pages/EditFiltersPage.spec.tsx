import React from "react";
import { render, store } from "../../test-utils";
import Filters from "./EditFiltersPage";
import { FilterType } from "../../types";

describe("EditFiltersPage", () => {
  it("renders", () => {
    const filter: FilterType = {
      name: "test",
      values: ["test"],
    };
    store.filters = [filter]; // @ts-ignore
    const { asFragment } = render(<Filters />);
    expect(asFragment()).toMatchSnapshot();
  });
});
