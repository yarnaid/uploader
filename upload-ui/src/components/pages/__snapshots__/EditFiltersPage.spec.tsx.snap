// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EditFiltersPage > renders 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <button
      class="btn btn-warning btn-sm"
      type="button"
    >
      New Filter(!)
    </button>
    <div />
  </div>
</DocumentFragment>
`;

exports[`EditFiltersPage renders 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <button
      class="btn btn-warning btn-sm"
      type="button"
    >
      New Filter(!)
    </button>
    <div />
  </div>
</DocumentFragment>
`;
